sched/scheduler/jobRunner.go

```diff
@@ -22,23 +22,22 @@ func runJob(job sched.Job, saga *s.Saga, nodes []cm.Node) {
 		return
 	}
 
-	dist := distributor.NewPoolDistributor(cm.StaticClusterFactory(nodes))
-
 	// if the saga has not been aborted, just run each task in the saga
 	// TODO: this can be made smarter to run the next Best Task instead of just
 	// the next one in order etc....
 	if !initialSagaState.IsSagaAborted() {
 		for id, task := range job.Def.Tasks {
-
+			fmt.Println("New task", id)
 			if !initialSagaState.IsTaskCompleted(id) {
-				node := dist.ReserveNode()
-
 				wg.Add(1)
-				go func(node cm.Node, taskId string, task sched.TaskDefinition) {
-					defer dist.ReleaseNode(node)
-					defer wg.Done()
-					runTask(saga, node, taskId, task)
-				}(node, id, task)
+				node := <-workers.Reserve
+				worker := (node.(worker.WorkerNode)).Worker()
```

Comment:
Yeah, what we really want is a Pool<Foo>, but go doesn't have generics.

We could make it just deal with WorkerNode, which moves this cast from jobRunner.go to poolDistributor.go (specifically, line 86, when we get the Node in from the NodeUpdate)